### PR TITLE
[MINOR] add sign up service

### PR DESCRIPTION
**Pull request by Mauriciopizarro**

### CODE REVIEW

**Comment by Mauriciopizarro** [comment]
@msotelo-eb i leave this pr for your code review.

File: controllers/sign_up_controller.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+from pydantic import BaseModel
+from fastapi import HTTPException, APIRouter
+from services.sign_up_service import SignUpService, EmptyPasswordError, UserExistent
```
Comment (matisotee) [inline controllers/sign_up_controller.py]: If these exceptions are used in several controllers, I would create a new file called `exceptions.py` in the controllers folder that contains all the exceptions for the controllers layer

File: controllers/sign_up_controller.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+sign_up_service = SignUpService()
+
+
+class SignUpRequestData(BaseModel):
+    username: str
+    password: str
+
+
+@router.post("/sign_up")
+async def sign_up_controller(json_data: SignUpRequestData):
```
Comment (matisotee) [inline controllers/sign_up_controller.py]: A better name for the first parameter would be `request` or `user`

File: controllers/sign_up_controller.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        raise HTTPException(
+            status_code=400,
+            detail="Empty password, please complete the password field",
+        )
+    except UserExistent:
+        raise HTTPException(
+            status_code=400,
+            detail="Username already in use, try another",
+        )
+    return {"User created succesfully"}
```
Comment (matisotee) [inline controllers/sign_up_controller.py]: We should return a token and the user id

File: services/sign_up_service.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+        if not plain_password:
+            raise EmptyPasswordError()
+
+        if self.user_repository.is_existent_user(username):
+            raise UserExistent()
+
+        database = self.user_repository.user_db
+        hashed_password = self.pwd_context.hash(plain_password)
+        database.add({"username": username, "hashed_password": hashed_password})
```
Comment (matisotee) [inline services/sign_up_service.py]: All this code is responsibility of the repository

**Review by matisotee — CHANGES_REQUESTED** [review]

**Pull request merged**